content-services/content-sources-backend
# Fixes 4645: show last error for templates

## Summary

- Adds new fields to the template response to track errors coming from the UpdateTemplateContent and UpdateLatestSnapshot tasks
- `last_update_snapshot_error` holds the last error coming from UpdateLatestSnapshot or an empty string if there isn't one
- `last_update_task` holds the response from the last UpdateTemplateContent task
- `last_update_task_uuid` holds the uuid from the last UpdateTemplateContent task

## Testing steps

1. Create a custom repository and let it snapshot
2. Create a template with that repo and `use_latest` set to true
3. Fetch the template. `last_update_snapshot_error` should be empty and `last_update_task` should hold the response from the latest UpdateTemplateContent task
4. Change the pulp URL in config.yaml to an invalid URL and edit the template. Fetching the template after this task completes all retries should show an error in `last_update_task` and a failed status
5. Force the UpdateLatestSnapshot task to fail (by either adding code to the Run method in update_latest_snapshot.go or making the pulp URL invalid as soon as a snapshot is completed). Fetch the template again and you should see an error in `last_update_snapshot_error` 
6. Change the pulp URL back to the valid URL and edit the template again to trigger the UpdateTemplateContent task. The `last_update_snapshot_error` field should now be cleared
